src/openloco/windows/IndustryWindow.cpp

```diff
@@ -0,0 +1,1064 @@
+                        {
+                            args.push(cargoObj->unit_name_singular);
+                        }
+                        else
+                        {
+                            args.push(cargoObj->unit_name_plural);
+                        }
+                        args.push(industry->received_cargo[i]);
+
+                        gfx::draw_string_left_wrapped(dpi, &origin, 290, colour::black, string_ids::white_stringid2, &args);
+                    }
+                }
+                origin.y += 4;
+            }
+
+            auto totalProducedCargo = industryObj->produced_cargo[0] & industryObj->produced_cargo[1];
+            // Draw Last Months produced cargo stats
+            if (totalProducedCargo != 0xFF)
+            {
+                gfx::draw_string_494B3F(*dpi, origin.x - 4, origin.y, colour::black, string_ids::produced_cargo);
+                origin.y += 10;
+
+                for (uint8_t i = 0; i < std::size(industryObj->produced_cargo); i++)
+                {
+                    uint16_t producedCargo = industryObj->produced_cargo[i];
+                    if (producedCargo != 0xFF)
+                    {
+                        auto cargoObj = objectmgr::get<cargo_object>(producedCargo);
+                        auto args = FormatArguments();
+
+                        if (producedCargo == 1)
+                        {
+                            args.push(cargoObj->unit_name_singular);
+                        }
+                        else
+                        {
+                            args.push(cargoObj->unit_name_plural);
+                        }
+                        args.push<uint32_t>(industry->produced_cargo[i]);
+                        args.push(industry->var_1A7[i]);
+
+                        gfx::draw_string_left_wrapped(dpi, &origin, 290, colour::black, string_ids::transported_cargo, &args);
+                    }
+                }
+            }
+        }
+
+        // 0x0045695E
+        static void on_mouse_up(window* self, widget_index widgetIndex)
+        {
+            switch (widgetIndex)
+            {
+                case common::widx::caption:
+                    common::renameIndustryPrompt(self, widgetIndex);
+                    break;
+
+                case common::widx::close_button:
+                    WindowManager::close(self);
+                    break;
+
+                case common::widx::tab_industry:
+                case common::widx::tab_production:
+                case common::widx::tab_production_2:
+                case common::widx::tab_transported:
+                    common::switchTab(self, widgetIndex);
+                    break;
+            }
+        }
+
+        // 0x004569C2
+        static void on_resize(window* self)
+        {
+            {
+                self->set_size(windowSize, windowSize);
+            }
+        }
+
+        // 0x0045698E
+        static void update(window* self)
```

Comment:
This can actually just be removed and the common function used instead. The frame_number is checked correctly when used and the invalidateWidget is just a more efficient invalidate. I suggest we remove this function and use the common version. Would be worth checking though to make sure that is true.

Reply:
I see no difference when using the common function so will remove it.